soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -176,6 +176,11 @@ void power_gate_entry(uint32_t core_id)
 	soc_cpus_active[core_id] = false;
 	sys_cache_data_flush_range(soc_cpus_active, sizeof(soc_cpus_active));
 	k_cpu_idle();
+
+	/* It is unlikely we get in here, but when this happens
+	 * we need to lock interruptions again.
+	 */
+	(void)irq_lock();
 	z_xt_ints_off(0xffffffff);
```

Comment:
Looks like this line is already masking all interrupts, though at one level higher in the abstraction stack (INTENABLE instead of PS.INTLVL).  So why do you need both?

Reply:
Because Zephyr only looks PS.INTENABLE and assumes that interruptions are not masked and triggers an assert.